testkit/src/main/java/org/apache/calcite/sql/test/AbstractSqlTester.java

```diff
@@ -325,7 +325,7 @@ public static String buildQueryAgg(String expression) {
    * @return Query that evaluates a scalar expression
    */
   protected String buildQuery2(SqlTestFactory factory, String expression) {
-    if (expression.matches("(?i).*(percentile_(cont|disc)|convert)\\(.*")) {
+    if (expression.matches("(?i).*(percentile_(cont|disc)|convert|sort_array)\\(.*")) {
       // PERCENTILE_CONT requires its argument to be a literal,
       // so converting its argument to a column will cause false errors.
```

Comment:
comments:
why should change this.
it is because it will change the literal to column for example select sort_array(array[1, 2], true) ->  sort_array(array[p0, p1], p2) . the p2 is not lieral